### PR TITLE
Finish rspec user and mailer

### PR DESCRIPTION
**Pull request by binhthanhnguyen**

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -5,7 +7,7 @@
     password "password"
     password_confirmation "password"
     admin false
-
```
Comment (trantube94) [inline spec/factories/users.rb]: remove

File: spec/factories/users.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# Read about factories at https://github.com/thoughtbot/factory_girl
+
```
Comment (trantube94) [inline spec/factories/users.rb]: remove

File: spec/mailers/previews/supervisor_mailer_preview.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Preview all emails at http://localhost:3000/rails/mailers/supervisor_mailer
+class SupervisorMailerPreview < ActionMailer::Preview
```
Comment (trantube94) [inline spec/mailers/previews/supervisor_mailer_preview.rb]: remove file

File: spec/mailers/supervisor_mailer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+RSpec.describe SupervisorMailer, type: :mailer do
+
+  describe ".monthly_report(supervisor)" do
+    before do
+      @admin = FactoryGirl.build(:admin)
+    end
+
+    it "sends monthly report from the default email" do
```
Comment (trantube94) [inline spec/mailers/supervisor_mailer_spec.rb]: "sends monthly report from the default email" ko viết cái này sau it

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
```
Comment (trantube94) [inline spec/models/user_spec.rb]: ''

File: spec/mailers/supervisor_mailer_spec.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    end
+
+    it "sends monthly report to admin" do
+      mail = SupervisorMailer.monthly_report @admin
+      expect(mail).to be_delivered_to(@admin.email)
+    end
+
+    it "has appropriate subject" do
+      mail = SupervisorMailer.monthly_report @admin
+      expect(mail).to have_subject("Monthly Report")
```
Comment (trantube94) [inline spec/mailers/supervisor_mailer_spec.rb]: remove ()

File: spec/mailers/supervisor_mailer_spec.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    end
+
+    it "sends monthly report from the default email" do
+      mail = SupervisorMailer.monthly_report @admin
+      expect(mail).to be_delivered_from("from@example.com")
+    end
+
+    it "sends monthly report to admin" do
+      mail = SupervisorMailer.monthly_report @admin
+      expect(mail).to be_delivered_to(@admin.email)
```
Comment (trantube94) [inline spec/mailers/supervisor_mailer_spec.rb]: remove ()

File: spec/mailers/supervisor_mailer_spec.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+RSpec.describe SupervisorMailer, type: :mailer do
+
+  describe ".monthly_report(supervisor)" do
+    before do
+      @admin = FactoryGirl.build(:admin)
+    end
+
+    it "sends monthly report from the default email" do
+      mail = SupervisorMailer.monthly_report @admin
+      expect(mail).to be_delivered_from("from@example.com")
```
Comment (trantube94) [inline spec/mailers/supervisor_mailer_spec.rb]: remove ()

File: spec/mailers/supervisor_mailer_spec.rb [hunk 9]
```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+RSpec.describe SupervisorMailer, type: :mailer do
+
+  describe ".monthly_report(supervisor)" do
+    before do
+      @admin = FactoryGirl.build(:admin)
```
Comment (trantube94) [inline spec/mailers/supervisor_mailer_spec.rb]: remove ()

File: spec/models/user_spec.rb [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+RSpec.describe User, type: :model do
+  it "is invalid without a email" do
+    user = FactoryGirl.build(:user, email: nil)
```
Comment (ttuan) [inline spec/models/user_spec.rb]: remove ()

File: spec/models/user_spec.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    user.valid?
+    expect(user.errors[:email]).to include "can't be blank"
+  end
+
+  it "is invalid without a password" do
+    user = FactoryGirl.build(:user, password: nil)
```
Comment (ttuan) [inline spec/models/user_spec.rb]: remove()

**Comment by VuongLong** [comment]
ok

**Comment by thiennv57** [comment]
OK

**Pull request merged**